XIPs/zip-15-attachment-content-type.md

```diff
@@ -0,0 +1,60 @@
+---
+xip: 15
+title: Attachment Content Type
+description: Content type for basic attachments
+author: Pat Nakajima
+status: Review
+type: Standards Track
+category: XRC
+created: 2023-02-15
+---
+
+## Abstract
+
+This XRC proposes a new content type for messages that do not consist of text. It attempts to include only the bare minimum for clients to determine how to display such messages in order to provide flexibility for more rich types in the future.
+
+## Motivation
+
+Not everything in a conversation is text. MMS, email attachments, gifs, etc are all examples of times when communications have gone beyond words. In order to provide users a real messaging platform, XMTP needs to provide this capability.
+
+## Specification
+
+Proposed content type:
+
+```js
+{
+  authorityId: "xmtp.org"
+  typeId: "attachment"
+  versionMajor: 0
+  versionMinor: 1
+}
+```
+
+The message MUST include the following parameters:
+
+```ts
+{
+  mimeType: string,
+  filename: string
+}
+```
```

Comment:
Some noob (I don't have a lot of MIME context) questions that I'm thinking about:

- What values are expected for mimeType and filename? Is the list of values for mimeType discrete or can it be anything? Should we reference something like the Mozilla list [here](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types).
- Should these strings be formatted in a particular way? If yes, how?
- What is the filename used for?
- How should frontend clients use the mimeType and filename info when rendering? The reference implementation is helpful but I'm wondering where frontends should follow a certain standard vs have the option to be opinionated?

Reply:
> What values are expected for mimeType and filename? Is the list of values for mimeType discrete or can it be anything? 

[MIME types](https://en.wikipedia.org/wiki/Media_type) are a pretty good standard for describing content types. It's possible someone could just put garbage in that field and clients wouldn't know what to do, but that's a possibility with any content description scheme. Clients should be able to handle corrupt messages for any number of reasons.

> Should these strings be formatted in a particular way? If yes, how?

It's just the type and subtype separated by a `/`. We can add some more examples here.

> What is the filename used for?

I added it because it's a good hint for what to name the file if a client wants to save it. It could also be used to display inline with the message.

> How should frontend clients use the mimeType and filename info when rendering?

To start out, they can look to see if the mime type is an image and show an image, if there's a type the client doesn't know how to render, they can do [a generic thing like iMessage](https://user-images.githubusercontent.com/483/219428312-1d132d74-4d63-4b9b-b1a7-c155c2e553f0.png).

> The reference implementation is helpful but I'm wondering where frontends should follow a certain standard vs have the option to be opinionated?

I think this is a case where clients can be opinionated. As we introduce more rich content types, we can try to give more guidance on usage.